File: Snackdown2k19/magic.cpp
#include<iostream>
#include<string>
#include<bits/stdc++.h>
#include <map>
#include <vector>
#include <unordered_set>
#include <iterator>



using namespace std;

int main()
{
	int t;
	cin>>t;

	while(t--)
	{
		int n;
		cin>>n;

		long long int a[n];
		long long partitions=0;

		for (int i = 0; i < n; ++i)
		{
			cin>>a[i];
			if(i>1)
			{
				if(a[i]<a[i-1])
					partitions++;
			}
		}

		if(partitions==0)
			cout<<"YES"<<endl;

		else if(partitions==1)
		{
			if(a[0] >= a[n-1])
				cout<<"YES"<<endl;

			else
				cout<<"NO"<<endl;
		}

		else
			cout<<"NO"<<endl;


	}
}

File: Snackdown2k19/Round1A/magic.py
t=int(input())
for i in range(t):
	n = int(input())

	l = [int(x) for x in input().split(" ")]

	partitions=0
	for i in range(len(l)-1):
		if(l[i] > l[i+1]):
			#print(l[i])
			partitions+=1

	#print(partitions)	
	if(partitions==0):
		print("YES")

	elif(partitions==1):
		if(l[0]>=l[len(l)-1]):
			print("YES")
		else:
			print("NO")
	
	else:
		print("NO")